EasyIntern-Backend/Controllers/CompanyController.cs

```diff
@@ -6,22 +6,47 @@
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
+
 namespace EasyIntern_Backend.Controllers
 {
     [Route("company"), Authorize]
     public class CompanyController : Controller
     {
         private readonly Context _context;
         private const int TakeAmount = 10;
+
         public CompanyController(Context context)
         {
             _context = context;
         }
+        [IsCompany]
+        [HttpGet("")]
+        public async Task<IActionResult> Index()
+        {
+            if (User.IsModerator()) //if the user is moderator return all companies
+            {
+                return Json(await _context.Users.Where(e => e.UserType == UserType.Company).ToListAsync());
+            }
+
+            if (!User.IsStudent())//user is company, return all students that have flirted with the users company.
```

Comment:
Why are we doing a check for `IsStudent`? There's already an `[IsCompany]` gaurd in place

Reply:
+1 (or ++, whatever code u prefer)